src/schemas/generator/index.ts

```diff
@@ -1,18 +1,22 @@
 import { z } from 'zod'
 import * as v0 from './v0'
 import * as v1 from './v1'
+import * as v2 from './v2'
 import { exhaustive } from '../../utils/typescript'
 
 const AnyGeneratorSchema = z.discriminatedUnion('version', [
   v0.GeneratorFileDataSchema,
   v1.GeneratorFileDataSchema,
+  v2.GeneratorFileDataSchema,
 ])
 
 export function migrate(generator: z.infer<typeof AnyGeneratorSchema>) {
   switch (generator.version) {
     case '0':
-      return migrate(v0.migrate(generator))
+      return migrate(v1.migrate(v0.migrate(generator)))
```

Comment:
`migrate` is a recursive function, so each step should only migrate the generator to the next version;
```ts
case '0':
  return migrate(v0.migrate(generator))
case '1.0':
  return migrate(v1.migrate(generator))
```